### ISSUE TITLE
make `pymmcore_plus.install` work for linux

### ISSUE THREAD

**Issue by tlambert03** [pymmcore-plus/pymmcore-plus#87]
@ianhi, this isn't something I've done before... how do you install for linux?

**Comment by ianhi** [comment]
also you can't get the secret device adapters, only the private ones :(

**Comment by ianhi** [comment]
part of why I want https://github.com/micro-manager/mmCoreAndDevices/issues/87

**Comment by tlambert03** [comment]
so you did!
gotcha.  we've got some amazing people building all these conda recipes for napari plugins these days... maybe I can bug them to do this

**Comment by ianhi** [comment]
It's a bit tricky because the build system there is pretty tightly coupled to the micro-manager build system. You can see about that here: https://github.com/micro-manager/mmCoreAndDevices/pull/86 and in particualr: https://github.com/micro-manager/mmCoreAndDevices/pull/86#issuecomment-1033224144 and https://github.com/micro-manager/micro-manager/issues/1392#issuecomment-1033140970

so the sad reality is probably that until those changes get made it's not oging to be easy to do on linux :(

**Comment by ianhi** [comment]
Although for CI at least all we really need is the DemoCamera `.so` file which for binder at least has been done and you can distribute like so: https://github.com/imjoy-team/micro-manager-imjoy/blob/42f982c8f360a6ba6a0c9f3d453ec81c2a7e48a5/postBuild

So we could compile that once and re-use in CI

**Comment by tlambert03** [comment]
thanks for your thoughts!

> On Linux, there is no single rule for "what the operating system provides", because that depends on the distribution (and what packages have been installed by the user).
> Conda presumably solves these issues by building its own world outside of the distro package manager (which is good and bad, depending on whether you live in that world or not).

yep. this is definitely one advantage of distributing with conda.  because it also distributes standard (non-python) libraries, one can declare all of the dependencies, test them in a minimal docker container, and make sure that they will come along with all of the needed C libraries, while avoiding binary incompatibilities.  I'd never advocate that you _force_ people to install with conda (which is the only bad part i can see with that scenario)... but there is definitely a huge part of the scientific community who might want to use mmcore that are "in" that world.  So once we get it working, it could be as simple as `conda install napari-micromanager` and you've got a full, platform-independent, python/C gui for micromanager.

anyway, I do see that you'll need to continue to develop/maintain all of these alternate approaches for maximum flexibility.  But I might also try to tag some folks to help us get it all building on conda.  For those in that world, it would definitely be convenient.

**Comment by marktsuchida** [comment]
Yes, I am aware that that is how Conda works, and I'm all for having Conda packages. Indeed, if you ignore the parts that require proprietary dependencies, Conda is the easiest way to build. In principle the problem of proprietary dependencies should be solvable with some sort of shim to access them when available, though it will take some (much?) work. Unless Conda has some mechanism for this situation that I'm not aware of.

But it's also not going to solve all our problems. There are at least a few (extreme) examples where licensing issues will get in the way. For example, there are DLLs that we are allowed to redistribute "as part of the application". Any mmCoreAndDevices-only distribution will not be able to include those (and no, the vendor will not provide them directly to users). There are other complicated cases that are less extreme. Some of these are prominent vendors. In any case, I don't really feel like checking every case and contacting the companies if the only end result is going to be that we have a second distribution that is identical except for excluding the Java app (even though I 100% agree that it would be nice to have). (There might be ways to strategize as a community to at least try to improve this situation fundamentally in the longer term.)

**Comment by tlambert03** [comment]
understood!  thanks again
